tests/phpunit/tests/media.php

```diff
@@ -4991,6 +4992,122 @@ public function test_wp_get_loading_optimization_attributes_low_res_image() {
 		);
 	}
 
+	/**
+	 * @ticket 58681
+	 *
+	 * @dataProvider data_wp_get_loading_optimization_attributes_in_shortcodes
+	 */
+	public function test_wp_get_loading_optimization_attributes_in_shortcodes( $setup, $expected, $message ) {
+		$attr = $this->get_width_height_for_high_priority();
+		$setup();
+
+		// The first image processed in a shortcode should have fetchpriority set to high.
+		$this->assertSame(
+			$expected,
+			wp_get_loading_optimization_attributes( 'img', $attr, 'do_shortcode' ),
+			$message
+		);
+	}
+
+	public function data_wp_get_loading_optimization_attributes_in_shortcodes() {
+		return array(
+			'default' => array(
```

Comment:
These keys normally describe the test case a little.

Reply:
Improved in [b9a8698](https://github.com/WordPress/wordpress-develop/pull/4799/commits/b9a869806907528ce361af7c0b9252dcf59ccb58)